File: 56. Merge Intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        LinkedList<int[]> res = new LinkedList<>();
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));

        for(int[] interval : intervals){
            if(res.isEmpty() || res.getLast()[1] < interval[0]){
                res.add(interval);
            }
            else{
                res.getLast()[1] = Math.max(res.getLast()[1], interval[1]);
            }
        }
        return res.toArray(new int[res.size()][]);
    }
}

File: python_july/56. Merge Intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x : x[0])
        n = len(intervals)
        i, curr = 1, intervals[0]
        res = []
        while i < n:
            if intervals[i][0] > curr[1]:
                res.append(curr)
                curr = intervals[i]
            else:
                curr = [curr[0], max(curr[1], intervals[i][1])]
            i += 1
        res.append(curr)
        return res